edxval/tests/test_api.py

```diff
@@ -269,6 +275,94 @@ def test_get_url_for_profile(self):
+            video=Video.objects.get(
+                edx_video_id=constants.VIDEO_DICT_FISH.get("edx_video_id")
+            ),
+            profile=Profile.objects.get(profile_name="desktop"),
+            **constants.ENCODED_VIDEO_DICT_DESKTOP
+        )
+        self.course_id = 'test-course'
+        CourseVideo.objects.create(video=video, course_id=self.course_id)
+        # 2nd video
+        video = Video.objects.create(**constants.VIDEO_DICT_STAR)
+        EncodedVideo.objects.create(
+            video=Video.objects.get(
+                edx_video_id=constants.VIDEO_DICT_STAR.get("edx_video_id")
+            ),
+            profile=Profile.objects.get(profile_name="mobile"),
+            **constants.ENCODED_VIDEO_DICT_MOBILE
+        )
+        EncodedVideo.objects.create(
+            video=Video.objects.get(
+                edx_video_id=constants.VIDEO_DICT_STAR.get("edx_video_id")
+            ),
+            profile=Profile.objects.get(profile_name="desktop"),
+            **constants.ENCODED_VIDEO_DICT_DESKTOP
+        )
+        CourseVideo.objects.create(video=video, course_id=self.course_id)
+
+    def test_get_video_for_course_profiles_success(self):
+        """
+        Tests series of successful get_video_info_for_course_and_profiles calls
+        """
+        videos = api.get_video_info_for_course_and_profiles(
+            'test-course',
+            ['mobile'])
+        self.assertEqual(len(videos[constants.VIDEO_DICT_FISH.get("edx_video_id")]), 1)
+        self.assertEqual(len(videos[constants.VIDEO_DICT_STAR.get("edx_video_id")]), 1)
+
+        videos = api.get_video_info_for_course_and_profiles(
+            'test-course',
+            ['mobile', 'desktop'])
+        self.assertEqual(len(videos[constants.VIDEO_DICT_FISH.get("edx_video_id")]), 2)
+        self.assertEqual(len(videos[constants.VIDEO_DICT_STAR.get("edx_video_id")]), 2)
+
+    def test_get_video_for_course_profiles_no_profile(self):
+        """Tests get_video_info_for_course_and_profiles with no such profile"""
+        videos = api.get_video_info_for_course_and_profiles(
+            'test-course',
+            ['no profile'])
+        self.assertEqual(len(videos), 0)
+
+        videos = api.get_video_info_for_course_and_profiles(
+            'test-course',
+            ['mobile', 'no_profile'])
+        self.assertEqual(len(videos[constants.VIDEO_DICT_FISH.get("edx_video_id")]), 1)
+        self.assertEqual(len(videos[constants.VIDEO_DICT_STAR.get("edx_video_id")]), 1)
+
+    def test_get_video_for_course_profiles_repeated_profile(self):
+        """
+        Tests get_video_info_for_course_and_profiles with repeated profile
+        """
+        videos = api.get_video_info_for_course_and_profiles(
+            'test-course',
+            ['mobile', 'mobile'])
+        self.assertEqual(len(videos[constants.VIDEO_DICT_FISH.get("edx_video_id")]), 1)
+        self.assertEqual(len(videos[constants.VIDEO_DICT_STAR.get("edx_video_id")]), 1)
+
```

Comment:
Too many blank lines here. Also, add test cases for specifying just one of the two profiles and for specifying an empty list of profiles